### ISSUE TITLE
Manage organizations per region

### ISSUE THREAD

**Issue by timoludwig** [digitalfabrik/integreat-cms#1566]
### Motivation

Instead of global organizations, we want them to be managed separately in each region.

### Proposed Solution

- Add region field to organization model
- Move organization form from network management to region area
- Only allow it to assign region users to organizations of their region
- Only allow it to assign pages to organizations of their region

**Comment by JanErikHuehne** [comment]
How do you want the region field of the organization model to be populated during migration? Do we want to keep it as empty (null) and retain an overview in the admin-dashboard of remaining not-region organizations? This might be removed later when the transitions has been made for all organizations ... ? Any other ideas?

**Comment by timoludwig** [comment]
> How do you want the region field of the organization model to be populated during migration? Do we want to keep it as empty (null) and retain an overview in the admin-dashboard of remaining not-region organizations? This might be removed later when the transitions has been made for all organizations ... ? Any other ideas?

Currently, we don't have any organizations in our production system, so this makes things a bit easier... We could just define this as backward incompatible change and delete all existing organizations with the note that they have to be re-added in the desired region.

> As for the assignment of users and pages: I think we have to do it dynamically right? At least in the admin-dashboard. Here we can select the preferred region for the user, the available organizations need to adjust to the choice here, right?

Yes, another option would be to move the assignment of users into the organization form... But I guess this makes less sense than selecting the organization in the user form. So yes, we'll need a little bit of JS so limit the choices of the organization field dynamically.

**Comment by JanErikHuehne** [comment]
So I have implemented most of it and pushed it into the branch. Works good so far. However, I am not very familiar with js though and need help implementing it. Could you help here? 

Another point: 
- So I did change the region field of the AbstractContentModel to be non required in order to be able to migrate. Otherwise, django will throw an error saying that this field needs to be populated during migration. I don´t think that this is very elegant. Is there a better way doing it?

**Comment by timoludwig** [comment]
> So I have implemented most of it and pushed it into the branch. Works good so far. However, I am not very familiar with js though and need help implementing it. Could you help here @timoludwig ?

Yes, please open a draft PR and I will have a look at it :slightly_smiling_face: 

> So I did change the region field of the AbstractContentModel to be non required in order to be able to migrate. Otherwise, django will throw an error saying that this field needs to be populated during migration. I don´t think that this is very elegant. Is there a better way doing it?

The `AbstractContentModel` is primarily meant for content types that have translations, and since there are too many differences between the `Page`/`Event`/`POI` models and the `Organization` model, we should stick with inheriting from the `AbstractBaseModel`, and just add a region field to the `Organization` model.
